contracts/upgradeable_contracts/erc20_to_erc20/HomeBridgeErcToErc.sol

```diff
@@ -129,13 +134,32 @@ contract HomeBridgeErcToErc is
         return 0xba4690f5; // bytes4(keccak256(abi.encodePacked("erc-to-erc-core")))
     }
 
-    function onExecuteAffirmation(address _recipient, uint256 _value, bytes32 txHash) internal returns (bool) {
+    /**
+     * @dev Internal callback to be called on successfull message execution.
+     * Should be called only after enough affirmations from the validators are already collected.
+     * @param _recipient address of the receiver where the new tokens should be minted.
+     * @param _value amount of tokens to mint.
+     * @param _txHash reference transaction hash on the Foreign side of the bridge which cause this operation.
+     * @param _hashMsg unique identifier of the particular bridge operation.
+     * @return true, if execution completed successfully.
+     */
+    function onExecuteAffirmation(address _recipient, uint256 _value, bytes32 _txHash, bytes32 _hashMsg)
+        internal
+        returns (bool)
+    {
+        (address aboveLimitsRecipient, uint256 aboveLimitsValue) = txAboveLimits(_hashMsg);
+        if (aboveLimitsRecipient != address(0)) {
+            // revert if a given transaction hash was already processed by the call to fixAssetsAboveLimits
+            require(aboveLimitsValue == _value);
+            setTxAboveLimits(address(0), 0, _hashMsg);
+            setOutOfLimitAmount(outOfLimitAmount().sub(_value));
+        }
```

Comment:
This code is the same in this file and in `HomeBridgeErcToNative.sol`. Does it make sense to have a separate method for this in `HomeOverdrawManagement`?

Reply:
Thanks. Added a `_clearAboveLimitsMarker` function.